examples/basic.cpp

```diff
@@ -0,0 +1,158 @@
+#include <cstdint>
+#include <cstdio>
+#include <cstring>
+
+#include <keyfinder/keyfinder.h>
+#include <keyfinder/audiodata.h>
+
+typedef struct WAVE_HEADER {
+    /* RIFF Chunk Descriptor */
+    uint8_t         ChunkID[4];
+    uint32_t        ChunkSize;
+    uint8_t         Format[4];
+    /* "fmt" sub-chunk */
+    uint8_t         SubChunk1ID[4];
+    uint32_t        SubChunk1Size;
+    uint16_t        AudioFormat;
+    uint16_t        NumChannels;
+    uint32_t        SampleRate;
+    uint32_t        ByteRate;
+    uint16_t        BlockAlign;
+    uint16_t        BitsPerSample;
+    /* "data" sub-chunk */
+    uint8_t         Subchunk2ID[4];
+    uint32_t        Subchunk2Size;
+} wave_header_t;
+
+// Static because it retains useful resources for repeat use
+static KeyFinder::KeyFinder k;
+
+int main(int argc, char* argv[]) {
+    if (argc != 2) {
+        fprintf(stderr, "usage: %s WAVE_FILE\n", argv[0]);
+        return 1;
+    }
+
+    const char* filePath = argv[1];
+    FILE* file = fopen(filePath, "r");
+    if (file == nullptr) {
+        fprintf(stderr, "Failed to open wave file: %s\n", filePath);
```

Comment:
```suggestion
        fprintf(stderr, "Failed to open WAVE file: %s\n", filePath);
```
for consistency

Reply:
otherwise LGTM

Reply:
fixed and force-pushed. please check again.